core/src/main/java/hudson/security/HudsonPrivateSecurityRealm.java

```diff
@@ -839,6 +839,12 @@ public Details newInstance(StaplerRequest2 req, JSONObject formData) throws Form
                     throw new FormException("Please confirm the password by typing it twice", "user.password2");
                 }
 
+                if (FIPS140.useCompliantAlgorithms()) {
+                    if (pwd.length()< FIPS_PASSWORD_LENGTH || pwd2.length() < FIPS_PASSWORD_LENGTH) {
```

Comment:
Maybe doing the check only in `user.password1` would make more sense? 
If lengths don't match the problem is that passwords are different and we would want the message in https://github.com/jenkinsci/jenkins/pull/9995/files#diff-18f9f27db9827aa6d004eb9000a9e27ec37197ae211d16346d0835f7c84fecd7R859 to be shown instead

Reply:
Tried fixing in recent commit.